bundles/framework/coordinatetool/plugin/CoordinatePluginHandler.js

```diff
@@ -0,0 +1,626 @@
+                        'lat': response.lat
+                    }
+                };
+
+                if (showMarker) {
+                    this.addMarker(newData, markerMessageData);
+                }
+
+                if (showMarker || centerMap) {
+                    this.centerMap(newData);
+                }
+
+                if (!centerMap) {
+                    this.updateLonLat(newData, true);
+                }
+            } else {
+                this.updateLonLat();
+            }
+        } catch (e) {
+            Messaging.error('display.cannotTransformCoordinates.message');
+        }
+        this.setLoading(false);
+    }
+
+    updateReverseGeocode (data) {}
+
+    async getEmergencyCallCoordinatesFromServer (data) {
+        // get the transform from current data
+        const sourceProjection = this.mapModule.getProjection();
+
+        // If coordinates are not  EPSG:4326 then
+        // need to get 'EPSG:4326' coordinates from service
+        if (sourceProjection !== 'EPSG:4326') {
+            try {
+                const response = await getTransformedCoordinatesFromServer(this.mapModule, data, sourceProjection, 'EPSG:4326');
+                if (response?.lat && response?.lon) {
+                    const newData = {
+                        'lonlat': {
+                            'lon': response.lon,
+                            'lat': response.lat
+                        }
+                    };
+                    return this.formatEmergencyCallMessage(newData);
+                }
+            } catch (e) {
+                Messaging.error(this.loc('display.cannotTransformCoordinates.message'));
+            }
+        }
+        // Else if coordinates are from 'EPSG:4326' then use these
+        else {
+            return this.formatEmergencyCallMessage(data);
+        }
+    }
+
+    async getEmergencyCallInfo (data) {
+        const xy = data || this.getMapXY();
+
+        // update emergency if configured
+        if (this.config.showEmergencyCallMessage) {
+            // already in degrees, don't fetch again
+            if (this.allowDegrees() && this.mapModule.getProjection() === 'EPSG:4326') {
+                this.updateState({
+                    emergencyInfo: this.formatEmergencyCallMessage({
+                        'lonlat': {
+                            'lon': xy.lonlat.lon,
+                            'lat': xy.lonlat.lat
+                        }
+                    })
+                });
+            } else {
+                await this.getEmergencyCallCoordinatesFromServer(data)
+                    .then(emergencyData => {
+                        this.updateState({
+                            emergencyInfo: emergencyData
+                        });
+                    });
+            }
+        } else {
+            this.updateState({
+                emergencyInfo: null
+            });
+        }
+        this.updatePopup();
```

Comment:
Config shouldn't change at runtime so we shouldn't need to updateState() on else nor updatePopup() if `showEmergencyCallMessage` is not configured.